filecoin-proofs/src/bin/paramcache.rs

```diff
@@ -52,19 +59,37 @@ fn cache_porep_params(porep_config: PoRepConfig) {
             StackedDrg<DefaultTreeHasher, DefaultPieceHasher>,
             _,
         >>::blank_circuit(&public_params);
-        let _ = StackedCompound::get_groth_params(circuit, &public_params);
+        let _ = if is_predictable {
+            StackedCompound::get_groth_params(
+                Some(&mut XorShiftRng::from_seed(SEED)),
+                circuit,
+                &public_params,
+            )
+        } else {
+            StackedCompound::get_groth_params(Some(&mut OsRng), circuit, &public_params)
+        }
+        .expect("failed to get groth params");
     }
     {
         let circuit = <StackedCompound as CompoundProof<
             _,
             StackedDrg<DefaultTreeHasher, DefaultPieceHasher>,
             _,
         >>::blank_circuit(&public_params);
-        let _ = StackedCompound::get_verifying_key(circuit, &public_params);
+        let _ = if is_predictable {
+            StackedCompound::get_verifying_key(
+                Some(&mut XorShiftRng::from_seed(SEED)),
+                circuit,
+                &public_params,
+            )
+        } else {
+            StackedCompound::get_verifying_key(Some(&mut OsRng), circuit, &public_params)
```

Comment:
Does this actually work? I think the verifying key and groth params need to be based on the same randomness, so they will need to be generated together. (If this works, I wonder why and maybe am missing something.)

Reply:
@porcuquine - I've resolved this. Good catch.